specifyweb/specify/migrations/0002_geo.py

```diff
@@ -0,0 +1,139 @@
+                'ordering': (),
+                'unique_together': (('collection', 'isdefault'),),
+            },
+        ),
+        migrations.AddField(
+            model_name='collectionobject',
+            name='cotype',
+            field=models.ForeignKey(db_column='COTypeID', null=True, on_delete=specifyweb.specify.models.protect_with_blockers, related_name='collectionobjects', to='specify.collectionobjecttype'),
+        ),
+        migrations.AddField(
+            model_name='collectionobject',
+            name='hasreferencecatalognumber',
+            field=models.BooleanField(blank=True, db_column='HasReferenceCatalogNumber', default=False, null=True),
+        ),
+        migrations.RunPython(create_default_collection_types), # reverse handeled by table deletion
+        migrations.CreateModel(
+            name='CollectionObjectGroup',
+            fields=[
+                ('id', models.AutoField(db_column='collectionobjectgroupid', primary_key=True, serialize=False)),
+                ('name', models.CharField(blank=True, db_column='Name', max_length=255, null=True)),
+                ('description', models.TextField(blank=True, db_column='Description', null=True)),
+                ('igsn', models.CharField(blank=True, db_column='IGSN', max_length=255, null=True)),
+                ('guid', models.CharField(blank=True, db_column='GUID', max_length=255, null=True)),
+                ('number1', models.SmallIntegerField(blank=True, db_column='Number1', null=True)),
+                ('version', models.IntegerField(blank=True, db_column='Version', default=0, null=True)),
+                ('timestampcreated', models.DateTimeField(db_column='TimestampCreated', default=django.utils.timezone.now)),
+                ('timestampmodified', models.DateTimeField(blank=True, db_column='TimestampModified', default=django.utils.timezone.now, null=True)),
+                ('text1', models.TextField(blank=True, db_column='Text1', null=True)),
+                ('text2', models.TextField(blank=True, db_column='Text2', null=True)),
+                ('text3', models.TextField(blank=True, db_column='Text3', null=True)),
+                ('collection', models.ForeignKey(db_column='CollectionID', on_delete=specifyweb.specify.models.protect_with_blockers, related_name='collectionobjectgroups', to='specify.collection')),
+                ('createdbyagent', models.ForeignKey(db_column='CreatedByAgentID', null=True, on_delete=specifyweb.specify.models.protect_with_blockers, related_name='+', to='specify.agent')),
+                ('modifiedbyagent', models.ForeignKey(db_column='ModifiedByAgentID', null=True, on_delete=specifyweb.specify.models.protect_with_blockers, related_name='+', to='specify.agent')),
+            ],
+            options={
+                'db_table': 'collectionobjectgroup',
+                'ordering': (),
+            },
+        ),
+        migrations.CreateModel(
+            name='CollectionObjectGroupJoin',
```

Comment:
I originally had the relationships for the CoJo table being like this, but was advised to change it to just parent/child
https://github.com/specify/specify7/blob/5b2e455887691bf3df9748594fdfd8596d8403fc/specifyweb/sp7_models/models.py#L93-L95

How's this:
```python
# Relationships: Many-to-One
cogparent = models.ForeignKey('CollectionObjectGroup', db_column='COGParentID', related_name='parentcojos', null=False, on_delete=models.CASCADE)

# Relationships: One-to-One
cogchild = models.OneToOneField('CollectionObjectGroup', db_column='COGChildID', related_name='childcojo', null=True, on_delete=models.CASCADE)
cochild = models.OneToOneField('CollectionObject', db_column='COChildID', related_name='cojo', null=True, on_delete=models.CASCADE)
```

In the future we can change it to many-to-one if we need to.  We could also try doing two separate join tables, one for COG-COG and one for COG-CO.  Or I guess we could forgo the CoJo table if we don't think we'll ever need it.

Reply:
>  but was advised to change it to just parent/child

Correct, bc adding that in the JOIN table didn't see fit. I think parent-child should be part of COG table (and not the join table) to prevent redundancy. See https://github.com/specify/specify7/pull/5032#discussion_r1651104244

Reply:
More simplified, the hierarchy (a COG is child of another COG) should be part of COG table (FK to itself)

COJO link to a COG (and they don't care and know that the related COG is a child or parent of another one), unlike the fks you currently have.

Just like determination (essentially the join table), collection object and taxon (essentially the cog table). Is this simpler to imagine?

Reply:
I think the idea from the geo-analyst meetings was that we might want to make the child relationship many-to-many in the future, so we want to set it up now in a way such that minimal changes will be needed in a possible future migration.

Reply:
need to discuss this later on, I don't get why the parent rel can't be on COG table, and why you want to have childcog?

First, I think name cogparent in COJO, cochild is misleading. It should be cog and co. that's it. 

COG:
parent -> COG (many-to-one)

COJO
cog -> COG (many-to-one)
co -> CO (many-to-one)

Why does this not support current use case, more simpler? Also supports many-to-many already from COG to CO. But, technically many-to-many not from COG to COG (one COG will have one parent, and multiple children). But, I don't think that'll be the usecase ever (even it is, this is the worst place to put it)